--- /usr/local/lib/python3.10/site-packages/supybot/plugins/BadWords/plugin.py
+++ /usr/local/lib/python3.10/site-packages/supybot/plugins/BadWords/plugin.py
@@ -82,8 +82,8 @@ class BadWords(callbacks.Privmsg):
                 if self.regexp.search(s):
                     c = irc.state.channels[channel]
                     cap = ircdb.makeChannelCapability(channel, 'op')
-                    if c.isHalfopPlus(irc.nick):
-                        if c.isHalfopPlus(msg.nick) or \
+                    if c.isVoicePlus(irc.nick):
+                        if c.isVoicePlus(msg.nick) or \
                                 ircdb.checkCapability(msg.prefix, cap):
                             self.log.debug("Not kicking %s from %s, because "
                                            "they are halfop+ or can't be "
--- /usr/local/lib/python3.10/site-packages/supybot/plugins/Services/plugin.py
+++ /usr/local/lib/python3.10/site-packages/supybot/plugins/Services/plugin.py
@@ -70,15 +70,14 @@ class Services(callbacks.Plugin):
     def __init__(self, irc):
         self.__parent = super(Services, self)
         self.__parent.__init__(irc)
+        network = irc.network if irc else None
+        for nick in self.registryValue('nicks', network=network):
+            config.registerNick(nick)
         self.reset()
 
     def reset(self):
         self.state = {}
 
-    def do001(self, irc, msg):
-        for nick in self.registryValue('nicks', network=irc.network):
-            config.registerNick(nick)
-
     def _getState(self, irc):
         return self.state.setdefault(irc.network, State())
